src/Captioning/Format/WebvttFile.php

```diff
@@ -6,98 +6,92 @@
 
 class WebvttFile extends File
 {
-    const TIMECODE_PATTERN = '#^([0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}) --> ([0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}) ?(.*)$#';
+    const TIMECODE_PATTERN = '#^((?:[0-9]{2}:)?[0-9]{2}:[0-9]{2}.[0-9]{3}) --> ((?:[0-9]{2}:)?[0-9]{2}:[0-9]{2}.[0-9]{3})( .*)?$#';
 
     protected $regions = array();
 
     public function parse()
```

Comment:
Parsing algorithm is completely rewritten and now it has a more rational form (in my opinion).